include/mbgl/util/lru_cache.hpp

```diff
@@ -0,0 +1,61 @@
+#pragma once
+
+#include <list>
+#include <stdexcept>
+#include <unordered_map>
+
+namespace mbgl {
+
+// Simple non-thread-safe LRU cache
+// Item must be hashable
+template <typename Item>
+class LRU {
+public:
+    // Number of cached items
+    size_t size() const { return list.size(); }
+
+    // Check if cache is empty
+    bool empty() const { return list.empty(); }
+
+    // Access x
+    // If x is already cached, move it to the front (most recently used)
+    void touch(Item x) {
+        auto it = map.find(x);
+        if (it != map.end()) {
+            list.splice(list.begin(), list, it->second);
+        } else {
+            list.push_front(x);
+            map[std::move(x)] = list.begin();
+        }
+    }
+
+    // Evict the least recently used item
+    // If the cache is empty, throw an exception
+    Item evict() {
+        if (list.empty()) {
+            throw std::runtime_error("LRU::evict: empty cache");
+        }
+        Item x = std::move(list.back());
+        map.erase(list.back());
+        list.pop_back();
+        return x;
+    }
+
+    // Check if item exists in cache without changing its order
+    bool isHit(Item x) { return map.find(std::move(x)) != map.end(); }
+
+    // remove item from cache if it exists
+    void remove(Item x) {
+        auto it = map.find(x);
+        if (it != map.end()) {
+            list.erase(it->second);
+            map.erase(it);
+        }
+    }
+
+private:
+    std::list<Item> list;
+    std::unordered_map<Item, typename std::list<Item>::iterator> map;
```

Comment:
Use `mbgl::unordered_map`.  I'd suggest allowing a custom hasher rather than requiring it to be injected into the `std` namespace.

I'm in the middle of doing that for something else, see:

https://github.com/maplibre/maplibre-native/pull/2703/files#diff-4480b4de146df8e7abe3bb167a73b03de99ced9d5e890b8dab5280993a1bbeaf

https://github.com/maplibre/maplibre-native/pull/2703/files#diff-703555c0df88e0fabf75a5bfd28acb970455cbd3abaec12e7f77bd7fb8333599

Reply:
done